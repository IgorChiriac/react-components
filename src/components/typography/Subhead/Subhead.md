```js
<Subhead>Subhead</Subhead>
```

```js
<Subhead style={{ textAlign: 'center', color: 'salmon' }}>Subhead</Subhead>
```
