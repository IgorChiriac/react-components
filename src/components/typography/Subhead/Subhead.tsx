import React from 'react';
import styled from 'styled-components';
import * as fonts from '../../../constants/fonts';

export interface ISubheadProps extends React.HTMLAttributes<HTMLParagraphElement> {}

const SSubhead = styled.p`
  font-weight: normal;
  font-size: 14px;
  line-height: 24px;
  letter-spacing: 0;
  font-family: ${fonts.openSans};
`;

const Subhead = (props: ISubheadProps) => {
  return <SSubhead {...props} />;
};

export default Subhead;
