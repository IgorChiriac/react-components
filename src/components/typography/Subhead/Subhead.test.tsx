import { axe } from 'jest-axe';
import React from 'react';
import { render } from 'react-testing-library';
import Subhead from './Subhead';

describe('<Subhead />', () => {
  it('renders the component with no a11y violations', async () => {
    const { container } = render(<Subhead>test</Subhead>);
    expect(container.firstChild).toMatchSnapshot();
    const results = await axe(container.innerHTML);
    expect(results).toHaveNoViolations();
  });
});
