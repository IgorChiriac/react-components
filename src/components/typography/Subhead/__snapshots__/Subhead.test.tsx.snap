// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Subhead /> renders the component with no a11y violations 1`] = `
.c0 {
  font-weight: normal;
  font-size: 14px;
  line-height: 24px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
}

<p
  class="c0"
>
  test
</p>
`;
